#3 Feat : 회원가입 - 회원정보 등록, 에러 해결

uri010 opened the pull request:
## 요약
- 회원가입 구현

## 변경 내용
- .gitignore 적용
- 회원가입 구현
    - 요청에 json 형태로 회원정보를 전달받으면 INSERT 문을 통해 "USER"테이블에 회원정보를 추가합니다.
- 에러 해결
    - application.properties에 `mybatis.mapper-locations`를 추가해 mapper의 위치를 지정해주어 mapper를 찾지 못하는 에러를 해결했습니다.
    - UserMapper.xml의 parameterType 옵션을 제거하고 not null로 설정된 invite_url에 임의의 값을 넣어주었습니다.
    - postman으로 요청을 보낼 때 "user_id" : "yulee"를 "userId" : "yulee"로 변경해 userId값이 null이 되는 문제를 해결했습니다. 


## Issue number/URL
#3 Feat : 회원가입 - 회원정보 등록

uri010 committed:
UserMapper.xml SQL 에러

uri010 committed:
#3 feat : .gitignore 적용

uri010 committed:
#3 feat : 회원가입 구현 및 에러 해결

ddl.sql의 테이블 생성 구문의 테이블 이름 "users"를 통일성 있게 "user"로 변경
UserMapper.xml에서 발생한 에러 해결
- 오류가 발생한 parameterType옵션 삭제
- not null인 invite_url에 임의의 값 할당
- created_at과 updated_at에도 NOW()로 값 할당

sunwootest commented on src/main/java/study/sharedcalendar/controller/UserController.java:
```diff
@@ -0,0 +1,21 @@
+    @PostMapping("")
+    public String signUp(@RequestBody User user) {
+        userService.signUp(user);
+        return "ok";
```
200 상태코드만 보내줘도 충분할 것 같은데 ok까지 클라이언트에게 전달해줄 필요가 있을까요? 조금 더 자세한 정보를 전달하거나 200코드만 보내줘도 충분해보입니다.

sunwootest commented on src/main/java/study/sharedcalendar/mapper/UserMapper.java:
```diff
@@ -0,0 +1,12 @@
+
+@Mapper
+public interface UserMapper {
+
+
```
불필요하게 공백이 존재하네요. 컨벤션을 지켜서 개발하는 것이 중요합니다 :)

sunwootest commented on src/main/java/study/sharedcalendar/service/UserService.java:
```diff
@@ -0,0 +1,17 @@
+public class UserService {
+    private final UserMapper userMapper;
+
+    public int signUp(User user) {
+        int result = userMapper.createUser(user);
+        return result;
+    }
```
회원가입을 할 때 꼭 int값을 반환해야할까요?

uri010 replied:
필요하지 않을 것 같습니다. 수정하겠습니다.

uri010 committed:
#3 fix : 리뷰사항 수정

controller의 signUp 메소드 리턴 타입을 void로 수정
공백 제거
service의 signUp 메소드 리턴 타입 void로 수정

uri010 commented:
리뷰 사항 수정했습니다.

sunwootest commented on src/main/java/study/sharedcalendar/controller/UserController.java:
```diff
@@ -0,0 +1,18 @@
+public class UserController {
+    private final UserService userService;
+
+    @PostMapping("")
```
("")는 생략해도됩니다.
